athenahealth/formurlencoder.go

```diff
@@ -0,0 +1,123 @@
+package athenahealth
+
+import (
+	"fmt"
+	"io"
+	"net/url"
+	"reflect"
+	"sort"
+	"strconv"
+)
+
+const (
+	defaultFormURLEncoderBufferSize = 512
+)
+
+type formURLEncoder struct {
+	entries map[string][]any
+}
+
+func NewFormURLEncoder() *formURLEncoder {
+	return &formURLEncoder{
+		entries: make(map[string][]any),
+	}
+}
+
+func (f *formURLEncoder) AddString(key string, value string) {
+	f.entries[key] = append(f.entries[key], value)
+}
+
+func (f *formURLEncoder) AddInt(key string, value int) {
+	f.entries[key] = append(f.entries[key], value)
+}
+
+func (f *formURLEncoder) AddReader(key string, value io.Reader) {
+	f.entries[key] = append(f.entries[key], value)
+}
+
+func (f *formURLEncoder) Encode(w io.Writer) error {
```

Comment:
Not sure if there's value in leaving a comment on the relationship between this and the url.Values `Encode()` ?

Reply:
added the same comment that `url.Values.Encode` has
> // Encode encodes the values into “URL encoded” form
// ("bar=baz&foo=quux") sorted by key.